### PR TITLE
Fix `nose` imports

### PR DESCRIPTION
**Pull request by mzient**
## Category:
**Bug fix** (*non-breaking change which fixes an issue*)

## Description:
We need to import `nose_utils` before anything directly using `nose` in Python 3.12.
Since Python 3.12 is not tested at pre-merge, it was easy to miss.
This PR replaces all direct imports of nose, nose.tool, nose.plugins, etc. so there's no bad example to copy from.

### Tests:

- [ ] Existing tests apply
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [X] N/A

### Documentation
- [ ] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [X] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [X] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

File: dali/test/python/reader/test_numpy.py [hunk 1]
```diff
@@ -23,7 +23,7 @@
 import random
 import tempfile
 from nose_utils import assert_raises
-from nose import SkipTest
+from nose_utils import SkipTest
```
Comment (JanuszL) [inline dali/test/python/reader/test_numpy.py:26]: Maybe you can merge it with the above.

File: dali/test/python/test_backend_impl.py [hunk 2]
```diff
@@ -23,7 +23,7 @@
 import nvidia.dali as dali
 
 from nose_utils import assert_raises
-from nose import SkipTest
+from nose_utils import SkipTest
```
Comment (JanuszL) [inline dali/test/python/test_backend_impl.py:26]: Maybe you can merge it with the above and below.

File: dali/test/python/checkpointing/test_dali_stateless_operators.py [hunk 3]
```diff
@@ -28,7 +28,7 @@
 )
 from nose2.tools import params, cartesian_params
 from nose_utils import assert_raises, SkipTest
-from nose.plugins.attrib import attr
+from nose_utils import attr
```
Comment (JanuszL) [inline dali/test/python/checkpointing/test_dali_stateless_operators.py:31]: Maybe you can merge it with the above.

File: dali/test/python/decoder/test_image.py [hunk 4]
```diff
@@ -279,7 +279,7 @@ def check_fancy_upsampling_body(batch_size, img_type, device):
 @params(1, 8)
 def test_fancy_upsampling(batch_size):
     if nvidia.dali.backend.GetNvjpegVersion() < 12001:
-        from nose import SkipTest
+        from nose_utils import SkipTest
```
Comment (JanuszL) [inline dali/test/python/decoder/test_image.py:282]: I would move this to the beginning of the file.

File: dali/test/python/decoder/test_jpeg_scan_limit.py [hunk 5]
```diff
@@ -19,7 +19,7 @@
 from nvidia.dali import pipeline_def
 
 from nose_utils import assert_raises
-from nose.plugins.attrib import attr
+from nose_utils import attr
```
Comment (JanuszL) [inline dali/test/python/decoder/test_jpeg_scan_limit.py:22]: Maybe you can merge it with the above.

File: dali/test/python/decoder/test_video.py [hunk 6]
```diff
@@ -23,8 +23,8 @@
 from test_utils import get_dali_extra_path, is_mulit_gpu, skip_if_m60
 from nvidia.dali.backend import TensorListGPU
 from nose2.tools import params
-from nose import SkipTest
-from nose.plugins.attrib import attr
+from nose_utils import SkipTest
+from nose_utils import attr
 from nose_utils import assert_raises
```
Comment (JanuszL) [inline dali/test/python/decoder/test_video.py:28]: Maybe you can merge it with the above.

File: dali/test/python/operator_1/test_laplacian.py [hunk 7]
```diff
@@ -21,7 +21,7 @@
 from scipy.ndimage import convolve1d, filters as sp_filters
 import os
 from nose_utils import assert_raises
-from nose.plugins.attrib import attr
+from nose_utils import attr
```
Comment (JanuszL) [inline dali/test/python/operator_1/test_laplacian.py:24]: Maybe you can merge it with the above.

File: dali/test/python/decoder/test_imgcodec.py [hunk 8]
```diff
@@ -280,7 +280,7 @@ def check_fancy_upsampling_body(batch_size, img_type, device):
 @params(1, 8)
 def test_fancy_upsampling(batch_size):
     if nvidia.dali.backend.GetNvjpegVersion() < 12001:
-        from nose import SkipTest
+        from nose_utils import SkipTest
```
[the remark this answers is not in the document]
Comment (mzient) [reply dali/test/python/decoder/test_imgcodec.py:283]: True.

File: dali/test/python/operator_2/test_random_object_bbox.py [hunk 9]
```diff
@@ -23,8 +23,7 @@
 import scipy.ndimage.measurements
 import random
 from test_utils import check_batch, np_type_to_dali
-from nose.tools import nottest
-from nose.plugins.attrib import attr
+from nose_utils import attr, nottest
```
Comment (JanuszL) [inline dali/test/python/operator_2/test_random_object_bbox.py:26]: Can be merged with L17

File: dali/test/python/reader/test_index.py [hunk 10]
```diff
@@ -23,7 +23,7 @@
 from test_utils import compare_pipelines, get_dali_extra_path
 from nose_utils import assert_raises, raises
 from nose2.tools import cartesian_params
-from nose import SkipTest
+from nose_utils import SkipTest
```
Comment (JanuszL) [inline dali/test/python/reader/test_index.py:26]: Can be merged with L24

File: dali/test/python/test_dali_tf_dataset_mnist_eager.py [hunk 11]
```diff
@@ -18,7 +18,7 @@
 import test_dali_tf_dataset_mnist as mnist
 from test_utils_tensorflow import skip_for_incompatible_tf, available_gpus
 from nose_utils import raises
-from nose import SkipTest
+from nose_utils import SkipTest
```
Comment (JanuszL) [inline dali/test/python/test_dali_tf_dataset_mnist_eager.py:21]: Maybe you can merge it with the above.

File: dali/test/python/test_external_source_impl_utils.py [hunk 12]
```diff
@@ -16,7 +16,7 @@
 from nvidia.dali import tensors, pipeline_def
 import nvidia.dali.fn as fn
 from nose_utils import raises, assert_equals
-from nose.plugins.attrib import attr
+from nose_utils import attr
```
Comment (JanuszL) [inline dali/test/python/test_external_source_impl_utils.py:19]: Maybe you can merge it with the above.

File: dali/test/python/test_functional_api.py [hunk 13]
```diff
@@ -20,7 +20,7 @@
 from nose_utils import assert_raises
 import sys
 import inspect
-from nose.plugins.attrib import attr
+from nose_utils import attr
```
Comment (JanuszL) [inline dali/test/python/test_functional_api.py:23]: Can be merged with L20

File: dali/test/python/test_optical_flow.py [hunk 14]
```diff
@@ -21,7 +21,7 @@
 from nvidia.dali import fn, types
 from test_utils import get_dali_extra_path, get_arch
 from nose_utils import raises, assert_raises
-from nose import SkipTest
+from nose_utils import SkipTest
```
Comment (JanuszL) [inline dali/test/python/test_optical_flow.py:24]: Maybe you can merge it with the above.

File: dali/test/python/test_pipeline_debug.py [hunk 15]
```diff
@@ -14,7 +14,7 @@
 
 import numpy as np
 import os
-from nose.plugins.attrib import attr
+from nose_utils import attr
```
Comment (JanuszL) [inline dali/test/python/test_pipeline_debug.py:17]: Can be merged with L23 and L25

File: dali/test/python/test_pipeline_inputs.py [hunk 16]
```diff
@@ -15,7 +15,7 @@
 
 import numpy as np
 import nvidia.dali.fn as fn
-from nose.plugins.attrib import attr
+from nose_utils import attr
```
Comment (JanuszL) [inline dali/test/python/test_pipeline_inputs.py:18]: Can be merged with L24

File: dali/test/python/test_torch_pipeline_rnnt.py [hunk 17]
```diff
@@ -23,7 +23,7 @@
 import math
 import random
 import os
-from nose.tools import nottest
+from nose_utils import nottest
```
Comment (JanuszL) [inline dali/test/python/test_torch_pipeline_rnnt.py:26]: Can be moved to L15

**Review by JanuszL — APPROVED** [review]

**Review by mdabek-nvidia — APPROVED** [review]

**Pull request merged**